### PR TITLE
Add hecdss Native API project

### PR DESCRIPTION
**Pull request by ktarbet**

### CODE REVIEW

File: heclib/hecdss/hecdss.c [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+    tss->allocated[zSTRUCT_TS_floatValues];
+    if (qualityArraySize > 0 && qualityArraySize == valueArraySize)
+      tss->quality = qualityArray;
+  }
+  else {
+    tss = zstructTsNewIrregDoubles(pathname, valueArray, valueArraySize, times,
+      timeGranularitySeconds, startDateBase, units, type);
+  }
+
+  const int storageFlag = 1;// storageFlag = 0  Always replace data (Regular Interval)
```
Comment (perrymanmd) [inline heclib/hecdss/hecdss.c:380]: See prev comment about enums.

File: heclib/hecdss/hecdss.c [hunk 2]
```diff
@@ -0,0 +1,742 @@
...
+      * retrieveFlag = 2 : Retrieve one value after end of time window
+      * retrieveFlag = 3 : Retrieve one value before and one value after time window
+      */
+    const int retrieveUsingTimeWindowFlag = 0; // Adhere to time window provided and generate the time array
+
+    int status = ztsRetrieve(dss->ifltab, tss, retrieveUsingTimeWindowFlag,retrieveDoublesFlag, boolRetrieveAnyQualityNotes);
+    if (status == 0) {
+        *julianBaseDate = tss->julianBaseDate;
+        *timeGranularitySeconds = tss->timeGranularitySeconds;
+
```
Comment (perrymanmd) [inline heclib/hecdss/hecdss.c:301]: Initialize char* out parameters below to empty strings either before `stringCopy(...)` or in `else` clauses.

File: heclib/hecdss/hecdss.c [hunk 3]
```diff
@@ -0,0 +1,742 @@
...
+    *x = loc->xOrdinate;
+    *y = loc->yOrdinate;
+    *z = loc->zOrdinate;
+    *coordinateSystem = loc->coordinateSystem;
+    *coordinateID = loc->coordinateID;
+    *verticalUnits = loc->verticalUnits;
+    *verticalDatum = loc->verticalDatum;
+    *horizontalUnits = loc->horizontalUnits;
+    *horizontalDatum = loc->horizontalDatum;
+
```
Comment (perrymanmd) [inline heclib/hecdss/hecdss.c:410]: Initialize char* out parameters below to empty strings either before `stringCopy(...)` or in `else` clauses.

File: heclib/hecdss/hecdss.c [hunk 4]
```diff
@@ -0,0 +1,742 @@
...
+                                    char* typeIndependent, const int typeIndependentLength,
+                                    char* typeDependent, const int typeDependentLength,
+                                    int* labelsLength){
+  zStructPairedData* pds = zstructPdNew(pathname);
+  //pds->endingOrdinate = 1;// hack to retrive meta data, with minimal actual data.
+  int returnDoubles = 2;
+  int status = zpdRetrieve(dss->ifltab, pds, returnDoubles);
+
+  *numberOrdinates = pds->numberOrdinates;
+  *numberCurves = pds->numberCurves;
```
Comment (perrymanmd) [inline heclib/hecdss/hecdss.c:492]: Initialize char* out parameters below to empty strings either before `stringCopy(...)` or in `else` clauses.

**Review by perrymanmd — CHANGES_REQUESTED** [review]

File: heclib/hecdss/hecdss.c [hunk 5]
```diff
@@ -0,0 +1,742 @@
...
+ 
+
+*/
+
+// private definition 
+struct dss_file {
+    long long ifltab[250];
+};
+
+HECDSS_API int hec_dss_test_modify_arg(int *value) {
```
[the remark this answers is not in the document]
Comment (ktarbet) [reply heclib/hecdss/hecdss.c:29]: Thanks - I'll remove these tests.

File: heclib/hecdss/hecdss.c [hunk 6]
```diff
@@ -0,0 +1,742 @@
...
+  return 0;
+}
+
+
+HECDSS_API int hec_dss_CONSTANT_MAX_PATH_SIZE() {
+  return MAX_PATHNAME_SIZE;
+}
+
+HECDSS_API int hec_dss_log_error(const char* message) {
+  printf("\nError %s", message);
```
[the remark this answers is not in the document]
Comment (ktarbet) [reply heclib/hecdss/hecdss.c:75]: Yes! but only as sophisticated as needed.   I'm open to ideas/help on that.

[the remark this answers is not in the document]
Comment (perrymanmd) [reply heclib/hecdss/hecdss.c:75]: All I was envisioning is something like:
```
int log_handle = stdout;
enum log_level {LOG_NONE, LOG_ERROR, LOG_WARNING};

HECDSS_API int hec_dss_log_error(const char* message) {
  if (log_level > LOG_NONE)
    fprintf(log_handle, "\nError %s", message);
  return 0;
}
HECDSS_API int hec_dss_log_warning(const char* message) {
  if (log_level > LOG_ERROR)
    fprintf(log_handle, "\nWarning: %s", message);
  return 0;
}
```
Then you could add functions to manage the output location and level if you wanted.

[the remark this answers is not in the document]
Comment (perrymanmd) [reply heclib/hecdss/hecdss.c:75]: And I'd be happy to help if/when you want to do it.

**Review by perrymanmd — APPROVED** [review]

**Pull request merged**